### ISSUE TITLE
Running mkver in puppet-dev-tools along with pdk

### ISSUE THREAD

**Issue by dhikrahashim** [idc101/git-mkver#42]
`git mkver next` is not incrementing the version. We are using the base [image](https://github.com/puppetlabs/puppet-dev-tools/blob/main/Dockerfile) to run the jenkins pipeline job.

1. First we need to increment the puppet module version using git-mkver next
2. Using that version we need to build the module using pdk build so we ahve used pdk based image. 

I believe image got some problem with git-mkver.

**Comment by idc101** [comment]
Do you have a bit more information? What config are you using? Did it work previously? What is in the git log? What version are you expecting and what version does it output?

**Comment by dhikrahashim** [comment]
Issue might be due to the container image we are running git mkver, it is Debian 11 based image and we are getting glibc error in this image and centos 7 too.

base image we have used is this one https://github.com/puppetlabs/puppet-dev-tools/blob/main/Dockerfile 

```
/tmp/git-mkver: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/git-mkver)
/tmp/git-mkver: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/git-mkver)
```

**Comment by dhikrahashim** [comment]
What config are you using?
 
 ```
tagPrefix: ""
defaults {
  whenNoValidCommitMessages: IncrementPatch
  patches: [
    metaData
  ]
}
patches: [
  {
    name: metaData
    filePatterns: ["metadata.json"]
    replacements: [
      {
        find: "\"version\": \"{VersionRegex}\""
        replace: "\"version\": \"{Next}\""
      }
    ]
  }
]
 ```
2. Did it work previously?

We are testing this first time for this puppet module code.

3. What is in the git log?

```
commit e2d2388943b8f57d892e8f67c6a2ad4a2bfa2b39   (HEAD -> master, origin/master, origin/HEAD)
Author: Removed intentionally
Date:   Tue Feb 20 13:59:15 2024 +0100

    minor: test2

commit 4e2b0d0d90eff6f0dd50435754f31396846ec960
Author: Removed intentionally
Date:   Tue Feb 20 13:57:41 2024 +0100

    feat: test
```

4. What version are you expecting and what version does it output?

It should increment the version based on the commit message or after each commit message, at least a patch version should get incremented. it always shows 0.1.0

metadata.json

```
{
  "name": "core-test",
  "version": "0.1.0",
  "author": "test@test.com",
  "summary": "Additional functions for Puppet",
  "license": "Apache-2.0",
  "source": "https://github.com/Dhi/puppet-core-test",
  "dependencies": [

  ],
  "operatingsystem_support": [
    {
      "operatingsystem": "RedHat",
      "operatingsystemrelease": [
        "6",
        "7"
      ]
    }
  ],
  "requirements": [
    {
      "name": "puppet",
      "version_requirement": "0.1.0"
    }
  ],
  "pdk-version": "0.1.0",
  "template-url": "pdk-default#0.1.0",
  "template-ref": "tags/0.1.0-0-g9a16c87"
}
```

**Comment by dhikrahashim** [comment]
Resolved the issue using version 1.2.2 and we should have the tag to work with git mkver. We missed.

**Comment by idc101** [comment]
Glad to hear you got it to work.
